scrape_data/cta_data_downloads.py

```diff
@@ -60,56 +60,166 @@ def save_csv_to_bucket(df: pd.DataFrame, filename: str) -> None:
+    if date_range is None:
+        date_range = [today]
+        print(f"No date range given. Using {today} only")
+            
+    start_date = pendulum.parse(min(date_range))
+    end_date = pendulum.parse(max(date_range))
+    period = pendulum.period(start_date, end_date)
+    full_date_range = [dt.to_date_string() for dt in period.range('days')]
+    zip_filename_list = [f'cta_schedule_zipfiles_raw/google_transit_{date}.zip'
+                         for date in full_date_range]
+
+    # Check for files in bucket.
+    found_list = keys(
+        csrt.BUCKET_PUBLIC,
+        zip_filename_list
     )
-    print(f'Confirm that {filename} exists in bucket')
-    keys(csrt.BUCKET_PUBLIC, [filename])
+    def confirm_saved_files(file_dict: dict) -> None:
+        for fname in ['csv_filenames', 'zip_filenames']:
+            print('Confirm that ' + ', '.join(file_dict[fname])
+                + ' exist in bucket')  
+            _ = keys(csrt.BUCKET_PUBLIC, file_dict[fname])
+    
+    def extract_date(fname: str) -> str:
+        return fname.split('_')[-1].split('.')[0]
+
+    def create_route_summary(CTA_GTFS: sga.GTFSFeed) -> pd.DataFrame:
+        data = sga.GTFSFeed.extract_data(CTA_GTFS)
+        data = sga.format_dates_hours(data)
+        trip_summary = sga.make_trip_summary(data)
+
+        route_daily_summary = (
+            sga.summarize_date_rt(trip_summary)
+        )
 
+        route_daily_summary['date'] = route_daily_summary['date'].astype(str)
+        route_daily_summary_today = route_daily_summary.loc[route_daily_summary['date'].isin(date_range)]
+        return route_daily_summary_today
+    
+    print('Using zipfiles found in public bucket')
+    s3zip_list = []
+    for fname in found_list:
+        zip_bytes = BytesIO()
+        zip_bytes.seek(0)
+        client.download_fileobj(Bucket=sga.BUCKET, Key=fname, Fileobj=zip_bytes)
+        zipfilesched = sga.zipfile.ZipFile(zip_bytes)
+        fdate = extract_date(fname)
+        s3zip_list.append(
+            {
+                'zip_filename': fname, 
+                'zip': zipfilesched,
+                'csv_filename': f'schedule_summaries/daily_job/'
+                                f'cta/cta_route_daily_summary_{fdate}.csv'
+            }
+        )    
+    
+    s3_route_daily_summary_dict = {
+        'zip_filenames': [gtfs['zip_filename'] for gtfs in s3zip_list],
+        'summaries': [create_route_summary(gtfs['zip']) for gtfs in s3zip_list],
+        'csv_filenames': [gtfs['csv_filename'] for gtfs in s3zip_list]
+    }
+
+    print(f'Saving cta schedule summary files in {date_range} to public bucket')
+    for filename, summary in zip(
+        s3_route_daily_summary_dict['csv_filenames'],
+        s3_route_daily_summary_dict['summaries']
+    ):
+        save_csv_to_bucket(summary, filename=filename)
+    
+    confirm_saved_files(s3_route_daily_summary_dict)
+
+    transitfeeds_list = list(set(zip_filename_list).difference(set(found_list)))
```

Comment:
This is the part that I think I would break out into like a one-time backfill... So instead of having this within the daily summaries, have like one function in a one-time script that checks for every date between 2022-05-20 and the present (maybe you can specify a smaller date range) and saves the zipfiles to S3. Maybe zipfiles from transitfeeds have a different name or something to distinguish them. 


And then all the daily summary stuff would only look for individual zipfiles in S3 and if a zipfile is not present it can just print an error and keep going and we can kind of fully decouple the zipfile downloading from the daily summary generation.... If that makes sense?